odl/solvers/nonsmooth/adupdates.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Alternating dual (AD) update algorithm studied by McGaffin and Fessler.
+
+The alternating dual upgrade method is reported to be a flexible method which
+allows an acceleration by permitting efficient calculations on a GPU.
+"""
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+__all__ = ('adupdates', 'adupdates_simple')
```

Comment:
The function `adupdates_simple` should not be exported.

Reply:
But if I do not export this function, how can I use it in tests or for debugging?

Reply:
It is still available under the full path `odl.solvers.nonsmooth.adaupdates.adaupdates_simple`.

Reply:
Technically, the `__all__` thing is for defining explicitly what should be exposed to wildcard imports, i.e. `from odl.solvers.nonsmooth.adaupdates import *`. If it's undefined, the wildcard import just imports everything not starting with an underscore (roughly).
For us, defining stuff in `__all__` is also a mechanism to make things available on a higher level, since the `__init__.py` files on higher levels do the wildcard import for all submodules (with exceptions).

So it's always possible to import names by going all the way down to where the name is actually defined (= bound to a value).